signature-envelope-cose.md

```diff
@@ -0,0 +1,197 @@
+- **`io.cncf.notary.expiry`** (*date/time*, critical): This OPTIONAL header provides a "best by use" time for the artifact, as defined by the signer. Its value is an Epoch-Based Date/Time defined in [RFC 8949](https://datatracker.ietf.org/doc/html/rfc8949#section-3.4.2).
+
+## Unprotected Headers
+
+Notary v2 supports the following unprotected header parameters:
+
+- `io.cncf.notary.timestampSignature`
+- Label `33`: `x5chain`
+- `io.cncf.notary.signingAgent`
+
+```yaml
+{
+  / x5chain / 33: [
+    << DER(leafCert) >>,
+    << DER(intermediate CACert) >>,
+    << DER(rootCert) >>
+  ],
+  'io.cncf.notary.timestampSignature': << TimeStampToken >>,
+  'io.cncf.notary.signingAgent': 'notation/1.0.0'
+}
+```
+
+Note: `<<` and `>>` are used to notate the CBOR byte string resulting from encoding the data item.
+
+- **[`x5chain`](https://datatracker.ietf.org/doc/html/draft-ietf-cose-x509-08#section-2)** (*array of bstr*): This REQUIRED parameter (label `33` by [IANA](https://www.iana.org/assignments/cose/cose.xhtml#header-parameters)) contains the ordered list of X.509 certificate or certificate chain ([RFC5280](https://datatracker.ietf.org/doc/html/rfc5280)) corresponding to the key used to digitally sign the COSE. The certificate chain is represented as an array of certificate, each certificate in the array is DER encoded and then wrapped in a CBOR byte string. The certificate containing the public key corresponding to the key used to digitally sign the COSE MUST be the first certificate, followed by the intermediate and root certificates in the correct order. Refer [*Certificate Chain* unsigned attribute](signature-specification.md#unsigned-attributes) for more details. Optionally, this header can be presented in the protected header.
+  - **TODO** update signature specification to allow chains in protected header
+- **`io.cncf.notary.timestampSignature`** (*bstr*): This OPTIONAL header is used to store countersignature that provides authentic signing time. Only [RFC3161]([rfc3161](https://datatracker.ietf.org/doc/html/rfc3161#section-2.4.2)) compliant `TimeStampToken` are supported.
+  - **TODO** Define the opaque datum (hash of envelope) that is sent to TSA, and how TSA response (time stamp token) is represented in this header.
+- **`io.cncf.notary.signingAgent`** (*tstr*): This OPTIONAL header provides the identifier of a client (e.g. Notation) that produced the signature. E.g. `notation/1.0.0`. Refer [*Signing Agent* unsigned attribute](signature-specification.md#unsigned-attributes) for more details.
+
+## Signature
+
+In COSE, signature is calculated by constructing the `Sig_structure` for `COSE_Sign1`.
+The process is described below:
+
+1. Encode the protected header into a CBOR object as a byte string named `body_protected`.
+2. Construct the `Sig_structure` for `COSE_Sign1`.
+    ```yaml
+    Sig_structure = [
+        / context / 'Signature1',
+        / body_protected / << ProtectedHeaders >>,
+        / external_aad / h'',
```

Comment:
What is `external_aad` , it's not defined here.

Reply:
The `external_aad` is the COSE external data.

Reply:
@gokarnm  Did shiwei answer your questions?

Reply:
Some content describing this field and if/how Notary v2 uses it, and any example will help.

Reply:
This field is explained here https://www.rfc-editor.org/rfc/rfc9052.html#Extern_AAD, since Notary v2 doesn't use it, so it's empty.